### PR TITLE
Migrating the DataSourceInterface

### PR DESCRIPTION
**Pull request by yaroslavborbat**
**What this PR does / why we need it**:
Added some public functions that link to private ones.
Methods causing panic have been removed from some datasources. (Filename(), Length(), ReadCloser())
The interface with these methods has been moved to dvcr-importer.

### CODE REVIEW

File: pkg/importer/transport.go [hunk 1]
```diff
@@ -102,6 +110,18 @@ func closeImage(src types.ImageSource) {
...
+}
+
+func IsWhiteout(path string) bool {
+	return isWhiteout(path)
+}
+
+func IsDir(hdr *tar.Header) bool {
+	return isDir(hdr)
+}
+
```
Comment (diafour) [inline pkg/importer/transport.go:124]: Let's move these public aliases to new file transport-public.go:

BuildSourceContext
ReadImageSource
HasPrefix
IsWhiteout
IsDir

Also, functions can be aliased as vars:

```
var BuildSourceContext = buildSourceContext
```

File: pkg/importer/upload-datasource.go [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 import (
-	"github.com/google/uuid"
 	"io"
 	"net/url"
 	"path/filepath"
 
+	"github.com/google/uuid"
+
 	"github.com/pkg/errors"
 	"k8s.io/klog/v2"
```
Comment (diafour) [inline pkg/importer/upload-datasource.go:11]:
```suggestion
	"github.com/google/uuid"
	"github.com/pkg/errors"
	"k8s.io/klog/v2"
```

File: pkg/importer/transport.go (hunk 1 above)
Comment (yaroslavborbat) [reply pkg/importer/transport.go]: You are right. I did it

File: pkg/importer/upload-datasource.go (hunk 2 above)
Comment (yaroslavborbat) [reply pkg/importer/upload-datasource.go]: ++

**Pull request merged**